crates/cext/src/circuit.rs

```diff
@@ -0,0 +1,264 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use std::ffi::{c_char, CString};
+
+use crate::exit_codes::ExitCode;
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::bit::{ShareableClbit, ShareableQubit};
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{Operation, Param, StandardGate};
+use qiskit_circuit::Qubit;
+
+/// @ingroup QkCircuit
+/// Construct a new circuit with the given number of qubits and clbits.
+///
+/// @param num_qubits The number of qubits the circuit contains.
+/// @param num_clbits The number of clbits the circuit contains.
+///
+/// @return A pointer to the created Circuit
```

Comment:
I'll leave some nits since it'll be the public docs 😛 
```suggestion
/// @return A pointer to the created circuit.
```